RELEASE.md

```diff
@@ -17,6 +17,41 @@ Please follow the established format:
 
 <!-- Add release notes for the upcoming release here -->
 
+# Release 3.3.0:
+
+## Major features and improvements
+
+- **BREAKING CHANGE:** Drop support for Python 3.5 (#138)
```

Comment:
While dropping  Python 3.5 is a major change, it's not a breaking change. This release won't break anything, but users using Python 3.5 will get Kedro-Viz 3.2.0 as latest

Reply:
Yeah I know, but I just thought I'd flag it anyway so that it's really noticeable?

Reply:
I think adding "Drop support for Python 3.5 (#138)" at the top of "Major features and improvements" is really noticeable :) I wouldn't say "BREAKING CHANGE" because it's not.